Tasks/JenkinsQueueJobV2/job.ts

```diff
@@ -437,7 +446,10 @@ export class Job {
                     thisJob.stopWork(0, JobState.Finishing);
                 }
             }
-        }).auth(thisJob.queue.TaskOptions.username, thisJob.queue.TaskOptions.password, true);
+        }).auth(thisJob.queue.TaskOptions.username, thisJob.queue.TaskOptions.password, true)
+        .on('error', (err) => { 
+            throw err; 
```

Comment:
I think that `throw new Error(err)` is a better pattern.  It will preserve the stacktrace.

Reply:
Shouldn't `err` already be an instance of the `Error` class?

Reply:
It probably is, but is it guaranteed to be?  Also, what do you want the stacktrace to look like?  I would want the top to be this line, and not something internal.

Reply:
I concur, converting callbacks to promises is something that we need to do to properly "fix" the task's error handling.  However in the interim, this fix will prevent builds from hanging when we lose the network connection with Jenkins while streaming the pipeline console output.  I have a story in the backlog to update the task to use promises.